### ISSUE TITLE
Add-AzureRmAccount : A task was canceled

### ISSUE THREAD

**Issue by GuofuZhong** [Azure/azure-powershell#6157]
Troubleshooting
Checkout how to troubleshoot failures and collect debug logs: https://docs.microsoft.com/en-us/vsts/build-release/actions/troubleshooting

Environment
VSTS
release definition:https://msasg.visualstudio.com/Bing_Ads/AdsUCM/_release?_a=releases&definitionId=447
release:
https://msasg.visualstudio.com/Bing_Ads/AdsUCM/_apps/hub/ms.vss-releaseManagement-web.cd-release-progress?releaseId=63514&_a=release-environment-logs&environmentId=346958
https://msasg.visualstudio.com/Bing_Ads/AdsUCM/_apps/hub/ms.vss-releaseManagement-web.cd-release-progress?releaseId=62994&_a=release-environment-logs&environmentId=345052

Agent - Hosted or Private:
private agent:Yes
version:2.129.1
OS: win server 2016.
Issue Description
we are using the VSTS multiplexer to run multiple azure db deployment scripts in parallel.
Sometimes we confront the exception:Add-AzureRmAccount : A task was canceled.

Error logs
Add-AzureRmAccount : A task was canceled.
At E:\VSTSAGENTVM022-008\_work\r2\a\Common\LoginAzureRM.ps1:60 char:5
+ &"Login-AzureRmAccount" @parameters |Out-Null
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo : CloseError: (:) [Add-AzureRmAccount], TaskCanceledException
+ FullyQualifiedErrorId : Microsoft.Azure.Commands.Profile.AddAzureRMAccountCommand

**Comment by rokmicrosoft** [comment]
This is same [issue ](https://github.com/Microsoft/vsts-tasks/issues/7049) created on azure powershell task team. Can you please look into this one why Login-AzureRmAccount is failing ?

**Comment by GuofuZhong** [comment]
we tested on one VM. if we run 6 jobs in parallel with calling Add-AzureRmAccount and same service principal. we always hit this error.

**Comment by markcowl** [comment]
@GuofuZhong Please provide the debug logs requested in the issue template.,.  Also, please provide your module layout on this machine:

```
gmo -ListAvailable
```

without this information, it will be difficult to rproduce or diagnose this issue.

**Comment by zgf** [comment]
@markcowl  add debug log
2018-05-09T00:15:22.6956489Z ##[debug]Evaluating condition for step: 'Apply Key Vault On Config'
2018-05-09T00:15:22.6957281Z ##[debug]Evaluating: succeeded()
2018-05-09T00:15:22.6957645Z ##[debug]Evaluating succeeded:
2018-05-09T00:15:22.6958585Z ##[debug]=> (Boolean) True
2018-05-09T00:15:22.6959883Z ##[debug]Expanded: True
2018-05-09T00:15:22.6960345Z ##[debug]Result: True
2018-05-09T00:15:22.6960867Z ##[section]Starting: Apply Key Vault On Config
2018-05-09T00:15:22.6967223Z ==============================================================================
2018-05-09T00:15:22.6967550Z Task         : PowerShell
2018-05-09T00:15:22.6967800Z Description  : Run a PowerShell script
2018-05-09T00:15:22.6968387Z Version      : 1.2.3
2018-05-09T00:15:22.6968617Z Author       : Microsoft Corporation
2018-05-09T00:15:22.6968911Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkID=613736)
2018-05-09T00:15:22.6969233Z ==============================================================================
2018-05-09T00:15:22.7018358Z ##[debug]Working directory: 'E:\VSTSAGENTVM021-008\_work\145\b'
2018-05-09T00:15:22.7020151Z ##[debug]C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoLogo -Sta -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command "try { $null = [System.Security.Cryptography.ProtectedData] } catch { Write-Verbose 'Adding assemly: System.Security' ; Add-Type -AssemblyName 'System.Security' ; $null = [System.Security.Cryptography.ProtectedData] ; $Error.Clear() } ; Invoke-Expression -Command ([System.Text.Encoding]::UTF8.GetString([System.Security.Cryptography.ProtectedData]::Unprotect([System.Convert]::FromBase64String('AQAAANCMnd8BFdERjHoAwE/Cl+sBAAAAXE6L7LK370OF7xN9PjIesgAAAAACAAAAAAADZgAAwAAAABAAAACZx6mv1WOxsvIwyUmG0qRfAAAAAASAAACgAAAAEAAAADqI/M6bhqwDcYlaRvzhxf7oAQAAdAyvB2+oHZFeCIwwBxhINJFkLTYozrzV1avwgeB3W/JNSRT9rZom7LG7elMc8laszWAi+VBhsoEgdoFl3dYz204pAGpPeLk4EY6eyDlBwSEie6CsdtPHCO6gzecwME5xRUFbEuNoEMaMwU51CmbDlRUDtkDxtQz5PzZrobvhGP62fVIKQBpu8izZliyDoxelBgiqqvLgGi5XROY3/N1yJXdoTGR79yWktVP3rLXp90z6qj8GG460M3kzEW3hR8iyETS3lO1dlsb+382UiDaWobw/Qg8CtPl5wwmIdfWwSV7DH14sA91sylA2IrQm86GkK23PnWg1fFCQSawPDnvt5dl66As7ueMlGEl1HDVL1Xmmgczk1yGtoQdcER1JGt7SvqfX1Qz3BKcKZhem9sNo1fOS2gnKqk+WkMz0kgjHjd5zjFbuhqvApGNmjIzxkz1wwxYIS/t3lIvdIdyZZDEciOW4mMN9XVeJy30SWdGZMRdr2ucuqHeFITrXRfZhaWqCBiOIpDGd63JtXBdKx6zPTun43CIaORbY2BxZz47wXUc6Ci+zCMwUNapz2DCJIPvsqziguxGF5Dyl2ZqZzMt49jeaBsVl8xdEA3KUXQz0zmRYwTCeoARlVGQQKgGMmmmp9nClAJemmv0UAAAALHeLV9a19nC+qHSFau/Oz2v09FU='), [System.Convert]::FromBase64String('wU3NkgJQTiq+wxoA8FtPwA=='), [System.Security.Cryptography.DataProtectionScope]::CurrentUser))) ; if (!(Test-Path -LiteralPath variable:\LastExitCode)) { Write-Verbose 'Last exit code is not set.' } else { Write-Verbose ('$LastExitCode: {0}' -f $LastExitCode) ; exit $LastExitCode }"
2018-05-09T00:15:22.7022669Z ##[command]. 'E:\VSTSAGENTVM021-008\_work\145\b\ConfigurationTesting\ApplyKeyVaultOnConfig.ps1' -username '********' -password '********' -tenantId '********' -devSubscriptionid '680a670d-5e26-424f-b524-3f4fb6b89319' -prodSubscriptionId 'fd074ab7-50d3-40ad-80d0-98bf53265400' -components "UCMWeb","UcmApi","UcmJobDispatcher" -workDirectory E:\VSTSAGENTVM021-008\_work\145\b -overwritingMapping '{}'
2018-05-09T00:15:57.5937451Z config save successfully:E:\VSTSAGENTVM021-008\_work\145\b\UCMWeb\Web.PROD.config
2018-05-09T00:17:57.1977020Z ##[error]Get-ChildItem : Get-KeyVaultSecret failed for: ucm-prod/ExchangeServiceUcmpremtPassword. Error Messsage: A task was 
canceled.
At E:\VSTSAGENTVM021-008\_work\145\b\ConfigurationTesting\ApplyKeyVaultOnConfig.ps1:106 char:13
+             Get-ChildItem -Path $sourceFolder |%{
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Get-KeyVaultSec...k was canceled.:String) [Get-ChildItem], RuntimeExc 
   eption
    + FullyQualifiedErrorId : Get-KeyVaultSecret failed for: ucm-prod/ExchangeServiceUcmpremtPassword. Error Messsage: 
    A task was canceled.,Microsoft.PowerShell.Commands.GetChildItemCommand
 

2018-05-09T00:17:58.0804004Z ##[error]Process completed with exit code 1 and had 1 error(s) written to the error stream.
2018-05-09T00:17:58.0822871Z ##[debug]System.Exception: Process completed with exit code 1 and had 1 error(s) written to the error stream.
   at Microsoft.VisualStudio.Services.Agent.Worker.Handlers.PowerShellExeHandler.<RunAsync>d__13.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Services.Agent.Worker.TaskRunner.<RunAsync>d__24.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Services.Agent.Worker.StepsRunner.<RunStepAsync>d__1.MoveNext()
2018-05-09T00:17:58.0827992Z ##[section]Finishing: Apply Key Vault On Config

**Comment by zgf** [comment]
https://msasg.visualstudio.com/Bing_Ads/_build/index?buildId=2181950&_a=summary

**Comment by zgf** [comment]
https://msasg.visualstudio.com/Bing_Ads/UCM%20Team/_apps/hub/ms.vss-releaseManagement-web.cd-release-progress?releaseId=63626&_a=release-environment-logs&environmentId=347425


2018-04-23T14:33:24.8801839Z ##[debug]Evaluating condition for step: 'UCM-PreloadKeyVault'
2018-04-23T14:33:24.8803305Z ##[debug]Evaluating: succeeded()
2018-04-23T14:33:24.8804335Z ##[debug]Evaluating succeeded:
2018-04-23T14:33:24.8805866Z ##[debug]=> (Boolean) True
2018-04-23T14:33:24.8806935Z ##[debug]Expanded: True
2018-04-23T14:33:24.8808047Z ##[debug]Result: True
2018-04-23T14:33:24.8809138Z ##[section]Starting: UCM-PreloadKeyVault
2018-04-23T14:33:24.8821280Z ==============================================================================
2018-04-23T14:33:24.8822130Z Task         : PowerShell
2018-04-23T14:33:24.8822900Z Description  : Run a PowerShell script
2018-04-23T14:33:24.8823610Z Version      : 1.2.3
2018-04-23T14:33:24.8824324Z Author       : Microsoft Corporation
2018-04-23T14:33:24.8825190Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkID=613736)
2018-04-23T14:33:24.8826159Z ==============================================================================
2018-04-23T14:33:24.9344082Z ##[debug]Working directory: 'E:\VSTSAGENTVM022-001\_work\r9\a'
2018-04-23T14:33:24.9350944Z ##[debug]C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoLogo -Sta -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command "try { $null = [System.Security.Cryptography.ProtectedData] } catch { Write-Verbose 'Adding assemly: System.Security' ; Add-Type -AssemblyName 'System.Security' ; $null = [System.Security.Cryptography.ProtectedData] ; $Error.Clear() } ; Invoke-Expression -Command ([System.Text.Encoding]::UTF8.GetString([System.Security.Cryptography.ProtectedData]::Unprotect([System.Convert]::FromBase64String('AQAAANCMnd8BFdERjHoAwE/Cl+sBAAAA/xV9suazgEmc0Tp/LgQ7HgAAAAACAAAAAAADZgAAwAAAABAAAAAPf3nkH5MXj7nauHHLTKQtAAAAAASAAACgAAAAEAAAAE53cCgJ9LEPULIXd/3A9y1oAAAAh5feBy+xtoGgiiRNpSZ7zLWcduX/Y/Hi1KZ7JQ3o6F8/HRTMNdEGMfla9D20DiRJZz/ysztbF4uyCfm5AHSn5n8GCyfZi15vU01EulafdsYQH/NAJfoaNVVLQA3VQhmWynmDA7fRZGcUAAAAJbk+ZMBNyudf4Rvow9YKFrhnqYY='), [System.Convert]::FromBase64String('l0DwiVWEH8ysfnY5HabYzw=='), [System.Security.Cryptography.DataProtectionScope]::CurrentUser))) ; if (!(Test-Path -LiteralPath variable:\LastExitCode)) { Write-Verbose 'Last exit code is not set.' } else { Write-Verbose ('$LastExitCode: {0}' -f $LastExitCode) ; exit $LastExitCode }"
2018-04-23T14:33:24.9356351Z ##[command]. 'C:\Users\ucmtest\AppData\Local\Temp\30e2595c-95d1-450d-8e7a-3884201bc477.ps1' prod 'VstsToken'
2018-04-23T14:33:25.2359036Z ==========Preload KeyVault==========
2018-04-23T14:33:25.2360152Z VaultName : ucm-prod
2018-04-23T14:33:25.2360968Z SubscriptionId : fd074ab7-50d3-40ad-80d0-98bf53265400
2018-04-23T14:35:09.1185819Z ##[error]Login-AzureRmAccount : A task was canceled.
At C:\Users\ucmtest\AppData\Local\Temp\30e2595c-95d1-450d-8e7a-3884201bc477.ps1:15 char:9
+         Login-AzureRmAccount -Credential $credential -ServicePrincipa ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Add-AzureRmAccount], TaskCanceledException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Profile.AddAzureRMAccountCommand
 
Set-AzureRmContext : Run Login-AzureRmAccount to login.
At C:\Users\ucmtest\AppData\Local\Temp\30e2595c-95d1-450d-8e7a-3884201bc477.ps1:48 char:5
+     Set-AzureRmContext -SubscriptionId $subscriptionId
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Set-AzureRmContext], PSInvalidOperationException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Profile.SetAzureRMContextCommand
 
Get-AzureKeyVaultSecret : Run Login-AzureRmAccount to login.
At C:\Users\ucmtest\AppData\Local\Temp\30e2595c-95d1-450d-8e7a-3884201bc477.ps1:68 char:9
+         Get-AzureKeyVaultSecret -VaultName $vaultName | Where-Object  ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Get-AzureKeyVaultSecret], PSInvalidOperationException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.KeyVault.GetAzureKeyVaultSecret
 

2018-04-23T14:35:09.1203345Z ==========Preload KeyVault successfully!==========
2018-04-23T14:35:09.1204241Z 103.8953158
2018-04-23T14:35:09.1653650Z ##[error]Process completed with exit code 0 and had 1 error(s) written to the error stream.
2018-04-23T14:35:09.1670979Z ##[debug]System.Exception: Process completed with exit code 0 and had 1 error(s) written to the error stream.
   at Microsoft.VisualStudio.Services.Agent.Worker.Handlers.PowerShellExeHandler.<RunAsync>d__13.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Services.Agent.Worker.TaskRunner.<RunAsync>d__24.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Services.Agent.Worker.StepsRunner.<RunStepAsync>d__1.MoveNext()
2018-04-23T14:35:09.1676093Z ##[section]Finishing: UCM-PreloadKeyVault

**Comment by zgf** [comment]
https://msasg.visualstudio.com/Bing_Ads/UCM%20Team/_apps/hub/ms.vss-releaseManagement-web.cd-release-progress?releaseId=63844&_a=release-environment-logs&environmentId=348366

2018-04-24T05:51:42.3355664Z ##[debug]Evaluating condition for step: 'Download ProdBits API FT Binaries from Azure Blob'
2018-04-24T05:51:42.3357761Z ##[debug]Evaluating: succeeded()
2018-04-24T05:51:42.3359078Z ##[debug]Evaluating succeeded:
2018-04-24T05:51:42.3361274Z ##[debug]=> (Boolean) True
2018-04-24T05:51:42.3364646Z ##[debug]Expanded: True
2018-04-24T05:51:42.3366165Z ##[debug]Result: True
2018-04-24T05:51:42.3367909Z ##[section]Starting: Download ProdBits API FT Binaries from Azure Blob
2018-04-24T05:51:42.3385424Z ==============================================================================
2018-04-24T05:51:42.3386501Z Task         : PowerShell
2018-04-24T05:51:42.3387457Z Description  : Run a PowerShell script
2018-04-24T05:51:42.3388370Z Version      : 1.2.3
2018-04-24T05:51:42.3389388Z Author       : Microsoft Corporation
2018-04-24T05:51:42.3390487Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkID=613736)
2018-04-24T05:51:42.3391716Z ==============================================================================
2018-04-24T05:51:42.3741723Z ##[debug]Working directory: 'E:\VSTSAGENTVM021-004\_work\r6\a'
2018-04-24T05:51:42.3751654Z ##[debug]C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoLogo -Sta -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command "try { $null = [System.Security.Cryptography.ProtectedData] } catch { Write-Verbose 'Adding assemly: System.Security' ; Add-Type -AssemblyName 'System.Security' ; $null = [System.Security.Cryptography.ProtectedData] ; $Error.Clear() } ; Invoke-Expression -Command ([System.Text.Encoding]::UTF8.GetString([System.Security.Cryptography.ProtectedData]::Unprotect([System.Convert]::FromBase64String('AQAAANCMnd8BFdERjHoAwE/Cl+sBAAAAXE6L7LK370OF7xN9PjIesgAAAAACAAAAAAADZgAAwAAAABAAAABgs0MRe0DFPIJDovZrm9QMAAAAAASAAACgAAAAEAAAANApFMVUViblwN044o76QFZYAgAApwAx5iHKqvSFo7xwPUtxeVAl2igU4Y+Pk18mL6RhPGJeVGY9aevOcPEwQmwFCyPeNJTE874kLLc+B9HXvZmkXPWiLEcY2dxnvoegBCFPXI2HD+FXAKLVmdnPDQ4VX43GsrN0psPYmU1szQgaorIusrweBw+xU6RI89AxgeQv2tpUrnWabjLVbcsqNI23w9FluWwoKaHtP0ysFE1m9dbQOAEGsWyU2k9WJzq93EnnektfYsTE4Olu+D/qcvM12H1hSSh10MChGvLqaJN+Dfm51fJUu7uKEvfbZA34CHd49ywFuO/FqhObUcxk3wNoPpt2A6qYWUPqdPXRLB0U3+i8ZLH3o6+tFZ7ZRL5ZSvOxDTE30aaWKdn/iBQO61qi86pl/qCz2WGZwLOt2ywqtlU6wmTzs/MweOiG8qEYdM3Mz1pUNKjxfKbPMObnHGxuOaekoNzOsIEFsv11cmiBydciQPeydvsdSUOpwVMv4NkAbWuugaAuNa75WjLUhxTs4OHxU5Uqw5ezkLI+KNiD7faESsmUAqJstA7RZMegiEbAmS6GVkTqfpNpRr0j53OZUNid1e+i0rb3JKQeIO6KnpG+PNGs+6TkW0kH/HohB211+geQ6040tOqVOqaykv88deytEjNrvoW7GtTJa/8BFKSEKrPmxQO5VaLzAMVhmyE/hViTFZs1mFzz+H/RaYe9aNzBBNvsI0CY/Ngdbb3ghyXazyrcWG8x/OXp67rz+Y6P9y3Ok0fsc1RtiAZKsCFK8ImQgOGIxGfkMqVqAZ2NHmqMlW0CsE/s803wFAAAABp7ff/qzCcm0j3pxw3uqhRzfbcx'), [System.Convert]::FromBase64String('mRouFaR5b85dvFLBMhNrMA=='), [System.Security.Cryptography.DataProtectionScope]::CurrentUser))) ; if (!(Test-Path -LiteralPath variable:\LastExitCode)) { Write-Verbose 'Last exit code is not set.' } else { Write-Verbose ('$LastExitCode: {0}' -f $LastExitCode) ; exit $LastExitCode }"
2018-04-24T05:51:42.3762712Z ##[command]. 'C:\Users\ucmtest.REDMOND\AppData\Local\Temp\0c8b27d8-3a03-42eb-85ae-727e8c4d5642.ps1' -AppAzureUser dc84cc07-dd1a-4881-ba45-af64c2f04045 -AppAzureKey ******** -AppAzureTenantId 72f988bf-86f1-41af-91ab-2d7cd011db47 -AzureResourceGroup mscorpnetvmsgroup -AzureStorageAccountName ucminfrastorage -AzureStorageAccountKey AZAJEakVrg5rOeeEkMTlKwBnmjwZGA0HR11pAYg6ndzi3x8OMU9pPG6DTVZdvuLYJyaKj95cLqVao5E9Yd5IKg== -LocalDestFolder E:\VSTSAGENTVM021-004\_work\r6\a\2067825\Binaries\retail\amd64\\UcmApi.FunctionalTests -ContainerPrefix apiftbackup -ReleaseId 63844
2018-04-24T05:53:25.2147077Z ##[error]Login-AzureRmAccount : A task was canceled.
At C:\Users\ucmtest.REDMOND\AppData\Local\Temp\0c8b27d8-3a03-42eb-85ae-727e8c4d5642.ps1:49 char:1
+ Login-AzureRmAccount -Credential $azureRmLogin.credential -ServicePri ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Add-AzureRmAccount], TaskCanceledException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Profile.AddAzureRMAccountCommand
 
Get-AzureRmStorageAccount : No account found in the context. Please login using Login-AzureRMAccount.
At C:\Users\ucmtest.REDMOND\AppData\Local\Temp\0c8b27d8-3a03-42eb-85ae-727e8c4d5642.ps1:51 char:20
+ ... eContext = (Get-AzureRmStorageAccount -StorageAccountName $AzureStora ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Get-AzureRmStorageAccount], ArgumentException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Management.Storage.GetAzureStorageAccountCommand
 
Get-AzureStorageContainer : Could not get the storage context.  Please pass in a storage context or set the current 
storage context.
At C:\Users\ucmtest.REDMOND\AppData\Local\Temp\0c8b27d8-3a03-42eb-85ae-727e8c4d5642.ps1:53 char:18
+ ... ainerList = Get-AzureStorageContainer -Context $storageContext -Prefi ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Get-AzureStorageContainer], InvalidOperationException
    + FullyQualifiedErrorId : InvalidOperationException,Microsoft.WindowsAzure.Commands.Storage.Blob.Cmdlet.GetAzureSt 
   orageContainerCommand
 

2018-04-24T05:53:25.2170160Z There is no existing backup with prefix apiftbackup
2018-04-24T05:53:25.2171459Z Uploading files of current release as new backup...
2018-04-24T05:53:29.7868049Z [2018/04/24 05:53:29] Transfer summary:
2018-04-24T05:53:29.7869618Z -----------------
2018-04-24T05:53:29.7870614Z Total files transferred: 567
2018-04-24T05:53:29.7871658Z Transfer successfully:   567
2018-04-24T05:53:29.7872652Z Transfer skipped:        0
2018-04-24T05:53:29.7873637Z Transfer failed:         0
2018-04-24T05:53:29.7874849Z Elapsed time:            00.00:00:04
2018-04-24T05:53:29.8814932Z ##[error]Process completed with exit code 0 and had 1 error(s) written to the error stream.
2018-04-24T05:53:29.8834876Z ##[debug]System.Exception: Process completed with exit code 0 and had 1 error(s) written to the error stream.
   at Microsoft.VisualStudio.Services.Agent.Worker.Handlers.PowerShellExeHandler.<RunAsync>d__13.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Services.Agent.Worker.TaskRunner.<RunAsync>d__24.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Services.Agent.Worker.StepsRunner.<RunStepAsync>d__1.MoveNext()
2018-04-24T05:53:29.8843513Z ##[section]Finishing: Download ProdBits API FT Binaries from Azure Blob

**Comment by yiqzhou** [comment]
Result of gmo-ListAvailable:

    Directory: C:\Program Files\WindowsPowerShell\Modules


ModuleType Version    Name                                ExportedCommands                                                                                               
---------- -------    ----                                ----------------                                                                                               
Script     5.0.0      Azure                               {Get-AzureAutomationCertificate, Get-AzureAutomationConnection, New-AzureAutomationConnection, Remove-AzureA...
Script     0.5.0      Azure.AnalysisServices              {Add-AzureAnalysisServicesAccount, Restart-AzureAnalysisServicesInstance, Export-AzureAnalysisServicesInstan...
Script     4.0.0      Azure.Storage                       {Get-AzureStorageTable, New-AzureStorageTableSASToken, New-AzureStorageTableStoredAccessPolicy, New-AzureSto...
Script     5.0.1      AzureRM                                                                                                                                            
Script     0.6.1      AzureRM.AnalysisServices            {Resume-AzureRmAnalysisServicesServer, Suspend-AzureRmAnalysisServicesServer, Get-AzureRmAnalysisServicesSer...
Script     0.5.0      AzureRM.AnalysisServices            {Resume-AzureRmAnalysisServicesServer, Suspend-AzureRmAnalysisServicesServer, Get-AzureRmAnalysisServicesSer...
Script     5.0.1      AzureRM.ApiManagement               {Add-AzureRmApiManagementRegion, Get-AzureRmApiManagementSsoToken, New-AzureRmApiManagementHostnameConfigura...
Script     0.1.0      AzureRM.ApplicationInsights         {Get-AzureRmApplicationInsights, New-AzureRmApplicationInsights, Remove-AzureRmApplicationInsights, Set-Azur...
Script     4.0.0      AzureRM.Automation                  {Get-AzureRMAutomationHybridWorkerGroup, Get-AzureRmAutomationJobOutputRecord, Import-AzureRmAutomationDscNo...
Script     4.0.1      AzureRM.Backup                      {Backup-AzureRmBackupItem, Enable-AzureRmBackupContainerReregistration, Get-AzureRmBackupContainer, Register...
Script     4.0.1      AzureRM.Batch                       {Remove-AzureRmBatchAccount, Get-AzureRmBatchAccount, Get-AzureRmBatchAccountKeys, New-AzureRmBatchAccount...} 
Script     0.14.0     AzureRM.Billing                     {Get-AzureRmBillingInvoice, Get-AzureRmBillingPeriod}                                                          
Script     4.0.0      AzureRM.Cdn                         {Get-AzureRmCdnProfile, Get-AzureRmCdnProfileSsoUrl, New-AzureRmCdnProfile, Remove-AzureRmCdnProfile...}       
Script     0.9.0      AzureRM.CognitiveServices           {Get-AzureRmCognitiveServicesAccount, Get-AzureRmCognitiveServicesAccountKey, Get-AzureRmCognitiveServicesAc...
Script     4.0.1      AzureRM.Compute                     {Remove-AzureRmAvailabilitySet, Get-AzureRmAvailabilitySet, New-AzureRmAvailabilitySet, Update-AzureRmAvaila...
Script     0.3.0      AzureRM.Consumption                 Get-AzureRmConsumptionUsageDetail                                                                              
Script     0.1.0      AzureRM.ContainerInstance           {New-AzureRmContainerGroup, Get-AzureRmContainerGroup, Remove-AzureRmContainerGroup, Get-AzureRmContainerIns...
Script     0.3.0      AzureRM.ContainerRegistry           {New-AzureRmContainerRegistry, Get-AzureRmContainerRegistry, Update-AzureRmContainerRegistry, Remove-AzureRm...
Script     4.0.1      AzureRM.DataFactories               {Remove-AzureRmDataFactory, Get-AzureRmDataFactoryRun, Get-AzureRmDataFactorySlice, Save-AzureRmDataFactoryL...
Script     0.3.0      AzureRM.DataFactoryV2               {Set-AzureRmDataFactoryV2, Get-AzureRmDataFactoryV2, Remove-AzureRmDataFactoryV2, Set-AzureRmDataFactoryV2Li...
Script     4.0.0      AzureRM.DataLakeAnalytics           {Get-AzureRmDataLakeAnalyticsDataSource, New-AzureRmDataLakeAnalyticsCatalogCredential, Remove-AzureRmDataLa...
Script     5.0.0      AzureRM.DataLakeStore               {Get-AzureRmDataLakeStoreTrustedIdProvider, Remove-AzureRmDataLakeStoreTrustedIdProvider, Remove-AzureRmData...
Script     4.0.0      AzureRM.DevTestLabs                 {Get-AzureRmDtlAllowedVMSizesPolicy, Get-AzureRmDtlAutoShutdownPolicy, Get-AzureRmDtlAutoStartPolicy, Get-Az...
Script     4.0.0      AzureRM.Dns                         {Get-AzureRmDnsRecordSet, New-AzureRmDnsRecordConfig, Remove-AzureRmDnsRecordSet, Set-AzureRmDnsRecordSet...}  
Script     0.2.0      AzureRM.EventGrid                   {New-AzureRmEventGridTopic, Get-AzureRmEventGridTopic, Set-AzureRmEventGridTopic, New-AzureRmEventGridTopicK...
Script     0.5.0      AzureRM.EventHub                    {New-AzureRmEventHubNamespace, Get-AzureRmEventHubNamespace, Set-AzureRmEventHubNamespace, Remove-AzureRmEve...
Script     4.0.1      AzureRM.HDInsight                   {Get-AzureRmHDInsightJob, New-AzureRmHDInsightSqoopJobDefinition, Wait-AzureRmHDInsightJob, New-AzureRmHDIns...
Script     4.0.0      AzureRM.Insights                    {Get-AzureRmMetricDefinition, Get-AzureRmMetric, Remove-AzureRmLogProfile, Get-AzureRmLogProfile...}           
Script     3.0.0      AzureRM.IotHub                      {Add-AzureRmIotHubKey, Get-AzureRmIotHubEventHubConsumerGroup, Get-AzureRmIotHubConnectionString, Get-AzureR...
Script     4.0.1      AzureRM.KeyVault                    {Add-AzureKeyVaultCertificate, Set-AzureKeyVaultCertificateAttribute, Stop-AzureKeyVaultCertificateOperation...
Script     4.0.0      AzureRM.LogicApp                    {Get-AzureRmIntegrationAccountAgreement, Get-AzureRmIntegrationAccountCallbackUrl, Get-AzureRmIntegrationAcc...
Script     0.16.0     AzureRM.MachineLearning             {Move-AzureRmMlCommitmentAssociation, Get-AzureRmMlCommitmentAssociation, Get-AzureRmMlCommitmentPlanUsageHi...
Script     0.2.0      AzureRM.MachineLearningCompute      {Get-AzureRmMlOpCluster, Get-AzureRmMlOpClusterKey, Test-AzureRmMlOpClusterSystemServicesUpdateAvailability,...
Script     0.2.0      AzureRM.MarketplaceOrdering         {Get-AzureRmMarketplaceTerms, Set-AzureRmMarketplaceTerms}                                                     
Script     0.8.0      AzureRM.Media                       {Sync-AzureRmMediaServiceStorageKeys, Set-AzureRmMediaServiceKey, Get-AzureRmMediaServiceKeys, Get-AzureRmMe...
Script     5.0.0      AzureRM.Network                     {Add-AzureRmApplicationGatewayAuthenticationCertificate, Get-AzureRmApplicationGatewayAuthenticationCertific...
Script     4.0.0      AzureRM.NotificationHubs            {Get-AzureRmNotificationHub, Get-AzureRmNotificationHubAuthorizationRules, Get-AzureRmNotificationHubListKey...
Script     4.0.0      AzureRM.OperationalInsights         {New-AzureRmOperationalInsightsAzureActivityLogDataSource, New-AzureRmOperationalInsightsCustomLogDataSource...
Script     4.0.0      AzureRM.PowerBIEmbedded             {Remove-AzureRmPowerBIWorkspaceCollection, Get-AzureRmPowerBIWorkspaceCollection, Get-AzureRmPowerBIWorkspac...
Script     4.1.1      AzureRM.profile                     {Disable-AzureRmDataCollection, Disable-AzureRmContextAutosave, Enable-AzureRmDataCollection, Enable-AzureRm...
Script     4.0.0      AzureRM.profile                     {Disable-AzureRmDataCollection, Disable-AzureRmContextAutosave, Enable-AzureRmDataCollection, Enable-AzureRm...
Script     4.0.1      AzureRM.RecoveryServices            {Get-AzureRmRecoveryServicesBackupProperty, Get-AzureRmRecoveryServicesVault, Get-AzureRmRecoveryServicesVau...
Script     4.0.1      AzureRM.RecoveryServices.Backup     {Backup-AzureRmRecoveryServicesBackupItem, Get-AzureRmRecoveryServicesBackupManagementServer, Get-AzureRmRec...
Script     0.2.1      AzureRM.RecoveryServices.SiteRec... {Edit-AzureRmRecoveryServicesAsrRecoveryPlan, Get-AzureRmRecoveryServicesAsrAlertSetting, Get-AzureRmRecover...
Script     4.0.1      AzureRM.RedisCache                  {Remove-AzureRmRedisCachePatchSchedule, New-AzureRmRedisCacheScheduleEntry, Get-AzureRmRedisCachePatchSchedu...
Script     0.3.0      AzureRM.Relay                       {New-AzureRmRelayNamespace, Get-AzureRmRelayNamespace, Set-AzureRmRelayNamespace, Remove-AzureRmRelayNamespa...
Script     5.0.0      AzureRM.Resources                   {Get-AzureRmProviderOperation, Remove-AzureRmRoleAssignment, Get-AzureRmRoleAssignment, New-AzureRmRoleAssig...
Script     0.16.0     AzureRM.Scheduler                   {Disable-AzureRmSchedulerJobCollection, Enable-AzureRmSchedulerJobCollection, Get-AzureRmSchedulerJobCollect...
Script     4.0.0      AzureRM.ServerManagement            {Invoke-AzureRmServerManagementPowerShellCommand, Get-AzureRmServerManagementSession, New-AzureRmServerManag...
Script     0.5.0      AzureRM.ServiceBus                  {New-AzureRmServiceBusNamespace, Get-AzureRmServiceBusNamespace, Set-AzureRmServiceBusNamespace, Remove-Azur...
Script     0.3.0      AzureRM.ServiceFabric               {Add-AzureRmServiceFabricApplicationCertificate, Add-AzureRmServiceFabricClientCertificate, Add-AzureRmServi...
Script     5.0.1      AzureRM.SiteRecovery                {Get-AzureRmSiteRecoveryFabric, New-AzureRmSiteRecoveryFabric, Remove-AzureRmSiteRecoveryFabric, Stop-AzureR...
Script     4.0.1      AzureRM.Sql                         {Get-AzureRmSqlDatabaseTransparentDataEncryption, Get-AzureRmSqlDatabaseTransparentDataEncryptionActivity, S...
Script     4.0.1      AzureRM.Storage                     {Get-AzureRmStorageAccount, Get-AzureRmStorageAccountKey, New-AzureRmStorageAccount, New-AzureRmStorageAccou...
Script     4.0.1      AzureRM.StreamAnalytics             {Get-AzureRmStreamAnalyticsFunction, Get-AzureRmStreamAnalyticsDefaultFunctionDefinition, New-AzureRmStreamA...
Script     4.0.0      AzureRM.Tags                        {Remove-AzureRmTag, Get-AzureRmTag, New-AzureRmTag}                                                            
Script     4.0.0      AzureRM.TrafficManager              {Disable-AzureRmTrafficManagerEndpoint, Enable-AzureRmTrafficManagerEndpoint, Set-AzureRmTrafficManagerEndpo...
Script     4.0.0      AzureRM.UsageAggregates             Get-UsageAggregates                                                                                            
Script     4.0.0      AzureRM.Websites                    {Get-AzureRmAppServicePlan, Set-AzureRmAppServicePlan, New-AzureRmAppServicePlan, Remove-AzureRmAppServicePl...
Script     1.0.0.1    DockerMsftProvider                                                                                                                                 
Script     1.0.1      Microsoft.PowerShell.Operation.V... {Get-OperationValidation, Invoke-OperationValidation}                                                          
Binary     1.0.0.1    PackageManagement                   {Find-Package, Get-Package, Get-PackageProvider, Get-PackageSource...}                                         
Script     3.4.0      Pester                              {Describe, Context, It, Should...}                                                                             
Script     1.0.0.1    PowerShellGet                       {Install-Module, Find-Module, Save-Module, Update-Module...}                                                   
Script     1.2        PSReadline                          {Get-PSReadlineKeyHandler, Set-PSReadlineKeyHandler, Remove-PSReadlineKeyHandler, Get-PSReadlineOption...}     
Manifest   21.0.17199 SqlServer                           {Add-SqlColumnEncryptionKeyValue, Complete-SqlColumnMasterKeyRotation, Get-SqlColumnEncryptionKey, Get-SqlCo...


    Directory: C:\Windows\system32\WindowsPowerShell\v1.0\Modules


ModuleType Version    Name                                ExportedCommands                                                                                               
---------- -------    ----                                ----------------                                                                                               
Manifest   1.0.0.0    AppBackgroundTask                   {Disable-AppBackgroundTaskDiagnosticLog, Enable-AppBackgroundTaskDiagnosticLog, Set-AppBackgroundTaskResourc...
Manifest   2.0.0.0    AppLocker                           {Get-AppLockerFileInformation, Get-AppLockerPolicy, New-AppLockerPolicy, Set-AppLockerPolicy...}               
Manifest   1.0.0.0    AppvClient                          {Add-AppvClientConnectionGroup, Add-AppvClientPackage, Add-AppvPublishingServer, Disable-Appv...}              
Manifest   2.0.0.0    Appx                                {Add-AppxPackage, Get-AppxPackage, Get-AppxPackageManifest, Remove-AppxPackage...}                             
Script     1.0.0.0    AssignedAccess                      {Clear-AssignedAccess, Get-AssignedAccess, Set-AssignedAccess}                                                 
Manifest   1.0        BestPractices                       {Get-BpaModel, Get-BpaResult, Invoke-BpaModel, Set-BpaResult}                                                  
Manifest   1.0.0.0    BitLocker                           {Unlock-BitLocker, Suspend-BitLocker, Resume-BitLocker, Remove-BitLockerKeyProtector...}                       
Manifest   2.0.0.0    BitsTransfer                        {Add-BitsFile, Complete-BitsTransfer, Get-BitsTransfer, Remove-BitsTransfer...}                                
Manifest   1.0.0.0    BranchCache                         {Add-BCDataCacheExtension, Clear-BCCache, Disable-BC, Disable-BCDowngrading...}                                
Manifest   1.0.0.0    CimCmdlets                          {Get-CimAssociatedInstance, Get-CimClass, Get-CimInstance, Get-CimSession...}                                  
Manifest   1.0        ConfigCI                            {Get-SystemDriver, New-CIPolicyRule, New-CIPolicy, Get-CIPolicy...}                                            
Binary     1.0.0.0    Containers                          {Add-ContainerNetworkAdapter, Add-ContainerNetworkAdapterStaticMapping, Get-ContainerNetworkAdapterStaticMap...
Manifest   1.0        Defender                            {Get-MpPreference, Set-MpPreference, Add-MpPreference, Remove-MpPreference...}                                 
Manifest   1.0.0.0    DirectAccessClientComponents        {Disable-DAManualEntryPointSelection, Enable-DAManualEntryPointSelection, Get-DAClientExperienceConfiguratio...
Script     3.0        Dism                                {Add-AppxProvisionedPackage, Add-WindowsDriver, Add-WindowsCapability, Add-WindowsImage...}                    
Manifest   1.0.0.0    DnsClient                           {Resolve-DnsName, Clear-DnsClientCache, Get-DnsClient, Get-DnsClientCache...}                                  
Manifest   1.0.0.0    EventTracingManagement              {New-EtwTraceSession, Get-EtwTraceSession, Set-EtwTraceSession, Send-EtwTraceSession...}                       
Manifest   1.0.0.0    HgsClient                           {Get-HgsAttestationBaselinePolicy, Get-HgsClientConfiguration, Set-HgsClientConfiguration, New-HgsGuardian...} 
Manifest   1.0.0.0    HgsDiagnostics                      {New-HgsTraceTarget, Get-HgsTrace, Get-HgsTraceFileData, Test-HgsTraceTarget}                                  
Manifest   1.0.0.0    HNVDiagnostics                      {Get-CustomerRoute, Get-PACAMapping, Get-ProviderAddress, Test-VirtualNetworkConnection...}                    
Binary     1.0.0.0    HostComputeService                  {Get-ComputeProcess, Stop-ComputeProcess}                                                                      
Binary     2.0.0.0    Hyper-V                             {Add-VMAssignableDevice, Add-VMDvdDrive, Add-VMFibreChannelHba, Add-VMGpuPartitionAdapter...}                  
Binary     1.1        Hyper-V                             {Add-VMDvdDrive, Add-VMFibreChannelHba, Add-VMHardDiskDrive, Add-VMMigrationNetwork...}                        
Manifest   2.0.0.0    International                       {Get-WinDefaultInputMethodOverride, Set-WinDefaultInputMethodOverride, Get-WinHomeLocation, Set-WinHomeLocat...
Manifest   1.0.0.0    iSCSI                               {Get-IscsiTargetPortal, New-IscsiTargetPortal, Remove-IscsiTargetPortal, Update-IscsiTargetPortal...}          
Manifest   2.0.0.0    IscsiTarget                         {Add-ClusteriSCSITargetServerRole, Add-IscsiVirtualDiskTargetMapping, Checkpoint-IscsiVirtualDisk, Convert-I...
Script     1.0.0.0    ISE                                 {New-IseSnippet, Import-IseSnippet, Get-IseSnippet}                                                            
Manifest   1.0.0.0    Kds                                 {Add-KdsRootKey, Get-KdsRootKey, Test-KdsRootKey, Set-KdsConfiguration...}                                     
Manifest   1.0.1.0    Microsoft.PowerShell.Archive        {Compress-Archive, Expand-Archive}                                                                             
Manifest   3.0.0.0    Microsoft.PowerShell.Diagnostics    {Get-WinEvent, Get-Counter, Import-Counter, Export-Counter...}                                                 
Manifest   3.0.0.0    Microsoft.PowerShell.Host           {Start-Transcript, Stop-Transcript}                                                                            
Manifest   1.0.0.0    Microsoft.PowerShell.LocalAccounts  {Add-LocalGroupMember, Disable-LocalUser, Enable-LocalUser, Get-LocalGroup...}                                 
Manifest   3.1.0.0    Microsoft.PowerShell.Management     {Add-Content, Clear-Content, Clear-ItemProperty, Join-Path...}                                                 
Script     1.0        Microsoft.PowerShell.ODataUtils     Export-ODataEndpointProxy                                                                                      
Manifest   3.0.0.0    Microsoft.PowerShell.Security       {Get-Acl, Set-Acl, Get-PfxCertificate, Get-Credential...}                                                      
Manifest   3.1.0.0    Microsoft.PowerShell.Utility        {Format-List, Format-Custom, Format-Table, Format-Wide...}                                                     
Manifest   3.0.0.0    Microsoft.WSMan.Management          {Disable-WSManCredSSP, Enable-WSManCredSSP, Get-WSManCredSSP, Set-WSManQuickConfig...}                         
Manifest   1.0        MMAgent                             {Disable-MMAgent, Enable-MMAgent, Set-MMAgent, Get-MMAgent...}                                                 
Manifest   1.0.0.0    MsDtc                               {New-DtcDiagnosticTransaction, Complete-DtcDiagnosticTransaction, Join-DtcDiagnosticResourceManager, Receive...
Manifest   2.0.0.0    NetAdapter                          {Disable-NetAdapter, Disable-NetAdapterBinding, Disable-NetAdapterChecksumOffload, Disable-NetAdapterEncapsu...
Manifest   1.0.0.0    NetConnection                       {Get-NetConnectionProfile, Set-NetConnectionProfile}                                                           
Manifest   1.0.0.0    NetEventPacketCapture               {New-NetEventSession, Remove-NetEventSession, Get-NetEventSession, Set-NetEventSession...}                     
Manifest   2.0.0.0    NetLbfo                             {Add-NetLbfoTeamMember, Add-NetLbfoTeamNic, Get-NetLbfoTeam, Get-NetLbfoTeamMember...}                         
Manifest   1.0.0.0    NetNat                              {Get-NetNat, Get-NetNatExternalAddress, Get-NetNatStaticMapping, Get-NetNatSession...}                         
Manifest   2.0.0.0    NetQos                              {Get-NetQosPolicy, Set-NetQosPolicy, Remove-NetQosPolicy, New-NetQosPolicy}                                    
Manifest   2.0.0.0    NetSecurity                         {Get-DAPolicyChange, New-NetIPsecAuthProposal, New-NetIPsecMainModeCryptoProposal, New-NetIPsecQuickModeCryp...
Manifest   1.0.0.0    NetSwitchTeam                       {New-NetSwitchTeam, Remove-NetSwitchTeam, Get-NetSwitchTeam, Rename-NetSwitchTeam...}                          
Manifest   1.0.0.0    NetTCPIP                            {Get-NetIPAddress, Get-NetIPInterface, Get-NetIPv4Protocol, Get-NetIPv6Protocol...}                            
Manifest   1.0.0.0    NetWNV                              {Get-NetVirtualizationProviderAddress, Get-NetVirtualizationGlobal, Get-NetVirtualizationLookupRecord, Get-N...
Manifest   1.0.0.0    NetworkConnectivityStatus           {Get-DAConnectionStatus, Get-NCSIPolicyConfiguration, Reset-NCSIPolicyConfiguration, Set-NCSIPolicyConfigura...
Manifest   1.0.0.0    NetworkSwitchManager                {Disable-NetworkSwitchEthernetPort, Enable-NetworkSwitchEthernetPort, Get-NetworkSwitchEthernetPort, Remove-...
Manifest   1.0.0.0    NetworkTransition                   {Add-NetIPHttpsCertBinding, Disable-NetDnsTransitionConfiguration, Disable-NetIPHttpsProfile, Disable-NetNat...
Manifest   1.0        NFS                                 {Get-NfsMappedIdentity, Get-NfsNetgroup, Install-NfsMappingStore, New-NfsMappedIdentity...}                    
Manifest   1.0.0.0    PcsvDevice                          {Get-PcsvDevice, Start-PcsvDevice, Stop-PcsvDevice, Restart-PcsvDevice...}                                     
Manifest   1.0.0.0    PKI                                 {Add-CertificateEnrollmentPolicyServer, Export-Certificate, Export-PfxCertificate, Get-CertificateAutoEnroll...
Manifest   1.0.0.0    PlatformIdentifier                  Get-PlatformIdentifier                                                                                         
Manifest   1.0.0.0    PnpDevice                           {Get-PnpDevice, Get-PnpDeviceProperty, Enable-PnpDevice, Disable-PnpDevice}                                    
Manifest   1.1        PrintManagement                     {Add-Printer, Add-PrinterDriver, Add-PrinterPort, Get-PrintConfiguration...}                                   
Manifest   1.1        PSDesiredStateConfiguration         {Set-DscLocalConfigurationManager, Start-DscConfiguration, Test-DscConfiguration, Publish-DscConfiguration...} 
Script     1.0.0.0    PSDiagnostics                       {Disable-PSTrace, Disable-PSWSManCombinedTrace, Disable-WSManTrace, Enable-PSTrace...}                         
Binary     1.1.0.0    PSScheduledJob                      {New-JobTrigger, Add-JobTrigger, Remove-JobTrigger, Get-JobTrigger...}                                         
Manifest   2.0.0.0    PSWorkflow                          {New-PSWorkflowExecutionOption, New-PSWorkflowSession, nwsn}                                                   
Manifest   1.0.0.0    PSWorkflowUtility                   Invoke-AsWorkflow                                                                                              
Manifest   2.0.0.0    RemoteDesktop                       {Get-RDCertificate, Set-RDCertificate, New-RDCertificate, New-RDVirtualDesktopDeployment...}                   
Manifest   1.0.0.0    ScheduledTasks                      {Get-ScheduledTask, Set-ScheduledTask, Register-ScheduledTask, Unregister-ScheduledTask...}                    
Manifest   2.0.0.0    SecureBoot                          {Confirm-SecureBootUEFI, Set-SecureBootUEFI, Get-SecureBootUEFI, Format-SecureBootUEFI...}                     
Manifest   1.0.0.0    SecurityCmdlets                     {Backup-SecurityPolicy, Restore-SecurityPolicy, Backup-AuditPolicy, Restore-AuditPolicy}                       
Script     1.0.0.0    ServerCore                          {Get-DisplayResolution, Set-DisplayResolution}                                                                 
Script     2.0.0.0    ServerManager                       {Get-WindowsFeature, Install-WindowsFeature, Uninstall-WindowsFeature, Enable-ServerManagerStandardUserRemot...
Cim        1.0.0.0    ServerManagerTasks                  {Get-SMCounterSample, Get-SMPerformanceCollector, Start-SMPerformanceCollector, Stop-SMPerformanceCollector...}
Manifest   2.0.0.0    SmbShare                            {Get-SmbShare, Remove-SmbShare, Set-SmbShare, Block-SmbShareAccess...}                                         
Manifest   2.0.0.0    SmbWitness                          {Get-SmbWitnessClient, Move-SmbWitnessClient, gsmbw, msmbw...}                                                 
Manifest   2.0.0.0    SoftwareInventoryLogging            {Get-SilComputer, Get-SilComputerIdentity, Get-SilSoftware, Get-SilWindowsUpdate...}                           
Manifest   1.0.0.0    StartLayout                         {Export-StartLayout, Import-StartLayout, Get-StartApps}                                                        
Manifest   2.0.0.0    Storage                             {Add-InitiatorIdToMaskingSet, Add-PartitionAccessPath, Add-PhysicalDisk, Add-TargetPortToMaskingSet...}        
Manifest   2.0.0.0    TLS                                 {New-TlsSessionTicketKey, Enable-TlsSessionTicketKey, Disable-TlsSessionTicketKey, Export-TlsSessionTicketKe...
Manifest   1.0.0.0    TroubleshootingPack                 {Get-TroubleshootingPack, Invoke-TroubleshootingPack}                                                          
Manifest   2.0.0.0    TrustedPlatformModule               {Get-Tpm, Initialize-Tpm, Clear-Tpm, Unblock-Tpm...}                                                           
Binary     2.1.639.0  UEV                                 {Clear-UevConfiguration, Clear-UevAppxPackage, Restore-UevBackup, Set-UevTemplateProfile...}                   
Manifest   1.0.0.0    UserAccessLogging                   {Enable-Ual, Disable-Ual, Get-Ual, Get-UalDns...}                                                              
Manifest   2.0.0.0    VpnClient                           {Add-VpnConnection, Set-VpnConnection, Remove-VpnConnection, Get-VpnConnection...}                             
Manifest   1.0.0.0    Wdac                                {Get-OdbcDriver, Set-OdbcDriver, Get-OdbcDsn, Add-OdbcDsn...}                                                  
Manifest   2.0.0.0    Whea                                {Get-WheaMemoryPolicy, Set-WheaMemoryPolicy}                                                                   
Manifest   1.0.0.0    WindowsDeveloperLicense             {Get-WindowsDeveloperLicense, Unregister-WindowsDeveloperLicense, Show-WindowsDeveloperLicenseRegistration}    
Script     1.0        WindowsErrorReporting               {Enable-WindowsErrorReporting, Disable-WindowsErrorReporting, Get-WindowsErrorReporting}                       
Manifest   1.0.0.0    WindowsSearch                       {Get-WindowsSearchSetting, Set-WindowsSearchSetting}                                                           
Manifest   1.0.0.0    WindowsUpdate                       Get-WindowsUpdateLog                                                                                           


    Directory: C:\Program Files\Microsoft Monitoring Agent\Agent\PowerShell


ModuleType Version    Name                                ExportedCommands                                                                                               
---------- -------    ----                                ----------------                                                                                               
Binary     1.0.0.0    Microsoft.MonitoringAgent.PowerS... {Checkpoint-WebApplicationMonitoring, Get-WebApplicationMonitoringStatus, Start-WebApplicationMonitoring, St...

**Comment by markcowl** [comment]
@GuofuZhong These are not debug logs for the actual azure powershell cmdlets, and so are useless for debugging the issue.  You will need to turn on the debug stream for the powershell session you are runnign these cmdlets in.

Also, it looks like you have unprotected data in this issue, which may have broadcast a credential.  Please ensure that this credential is no longer valid, as this could be a security breach.

This is a quite old version of AzureRM, have you tried upgrading to a more recent version?

**Comment by GuofuZhong** [comment]
@markcowl we have updated AzureRM to latest. but we still see this exception.
How to enbable debug stream for powershell session?

**Comment by GuofuZhong** [comment]
2018-05-14T05:27:00.5653823Z ##[debug]Evaluating condition for step: 'Apply Key Vault On Config'
2018-05-14T05:27:00.5655000Z ##[debug]Evaluating: succeeded()
2018-05-14T05:27:00.5655371Z ##[debug]Evaluating succeeded:
2018-05-14T05:27:00.5655935Z ##[debug]=> (Boolean) True
2018-05-14T05:27:00.5656378Z ##[debug]Expanded: True
2018-05-14T05:27:00.5656769Z ##[debug]Result: True
2018-05-14T05:27:00.5657237Z ##[section]Starting: Apply Key Vault On Config
2018-05-14T05:27:00.5664516Z ==============================================================================
2018-05-14T05:27:00.5664869Z Task         : PowerShell
2018-05-14T05:27:00.5665169Z Description  : Run a PowerShell script
2018-05-14T05:27:00.5665440Z Version      : 1.2.3
2018-05-14T05:27:00.5665752Z Author       : Microsoft Corporation
2018-05-14T05:27:00.5666096Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkID=613736)
2018-05-14T05:27:00.5666457Z ==============================================================================
2018-05-14T05:27:00.5731484Z ##[debug]Working directory: 'E:\VSTSAGENTVM022-004\_work\135\b'
2018-05-14T05:27:00.5733738Z ##[debug]C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoLogo -Sta -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command "try { $null = [System.Security.Cryptography.ProtectedData] } catch { Write-Verbose 'Adding assemly: System.Security' ; Add-Type -AssemblyName 'System.Security' ; $null = [System.Security.Cryptography.ProtectedData] ; $Error.Clear() } ; Invoke-Expression -Command ([System.Text.Encoding]::UTF8.GetString([System.Security.Cryptography.ProtectedData]::Unprotect([System.Convert]::FromBase64String('AQAAANCMnd8BFdERjHoAwE/Cl+sBAAAA/xV9suazgEmc0Tp/LgQ7HgAAAAACAAAAAAADZgAAwAAAABAAAADaGvKAJ3KCYKxMbzH9Xz6qAAAAAASAAACgAAAAEAAAANBiFjY96z+1INweDmG8k5ToAQAAqAGHtp+lxBcU8onYF9kYRlIeLROWlnoz7xvBJjoCnAGB5uqSYjmwoo6rev/cV2nmzZj1k+EuKLHiE15TG/94z91jla/cLgSEMHmuS5lvFmfUszC6cj5XPO/LT9SAv4ECfBSgt9oZe1EeHMSCK0TPh2WhINOoeDGrPDu1mlO6FM61Fet+lH4hdGMwLAo+Y+8Dz4wBVZi4curY9bpWD8Etv2GkOMqCqEH8BlKV7dHvS6KzYfoNUrR+s5RVMMxvyAGBJHmYq11ci6gPxt3o12Mdsz4SioftLOtycu/LipcfKqQZwe76XKYc1WEAva69xZkOrL7gAo5nJG3jtNj81bB80zHopYV8gsk+BbpAMncrtUKI86AXYhtD5JuoM8/6TSLwDPe7YktG6Y0OmIePBhX1TG/qxLUyZF/Ly9rqIXlAZwJy7GzOSNoZveXNG5DxMIoTHp4ha3Qx06mm7sEZX8rwHR8Yoqb0+dUmj+G/AWpLv7wci8KXx7EJ4tNVSRoeMk+PPZFTs3W85A9awIGwyYi750GrRp7IiZgIohlAvQAbJmvuZVrFGYGuW29CejmrjJ8fOH5ldmXsuLNoDJ7pGeWrIunH1/p3gVVedU0NWITiXTQVC6VDbgIIDAdQnrGEr0LtliDoKZesxgUUAAAAHQXdNmrACCo1EOUsakWhV+lWt1o='), [System.Convert]::FromBase64String('EEDPjGwejbzOjktywP3iHg=='), [System.Security.Cryptography.DataProtectionScope]::CurrentUser))) ; if (!(Test-Path -LiteralPath variable:\LastExitCode)) { Write-Verbose 'Last exit code is not set.' } else { Write-Verbose ('$LastExitCode: {0}' -f $LastExitCode) ; exit $LastExitCode }"
2018-05-14T05:27:00.5736330Z ##[command]. 'E:\VSTSAGENTVM022-004\_work\135\b\ConfigurationTesting\ApplyKeyVaultOnConfig.ps1' -username '********' -password '********' -tenantId '********' -devSubscriptionid '680a670d-5e26-424f-b524-3f4fb6b89319' -prodSubscriptionId 'fd074ab7-50d3-40ad-80d0-98bf53265400' -components "UCMWeb","UcmApi","UcmJobDispatcher" -workDirectory E:\VSTSAGENTVM022-004\_work\135\b -overwritingMapping '{}'
2018-05-14T05:27:14.3675827Z config save successfully:E:\VSTSAGENTVM022-004\_work\135\b\UCMWeb\Web.PROD.config
2018-05-14T05:29:23.4070079Z ##[error]Get-ChildItem : Get-KeyVaultSecret failed for: ucm-prod/ExchangeServiceUcmdmailPassword. Error Messsage: A task was 
canceled.
At E:\VSTSAGENTVM022-004\_work\135\b\ConfigurationTesting\ApplyKeyVaultOnConfig.ps1:106 char:13
+             Get-ChildItem -Path $sourceFolder |%{
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Get-KeyVaultSec...k was canceled.:String) [Get-ChildItem], RuntimeExc 
   eption
    + FullyQualifiedErrorId : Get-KeyVaultSecret failed for: ucm-prod/ExchangeServiceUcmdmailPassword. Error Messsage: 
    A task was canceled.,Microsoft.PowerShell.Commands.GetChildItemCommand
 

2018-05-14T05:29:23.6111883Z ##[error]Process completed with exit code 1 and had 1 error(s) written to the error stream.
2018-05-14T05:29:23.6129050Z ##[debug]System.Exception: Process completed with exit code 1 and had 1 error(s) written to the error stream.
   at Microsoft.VisualStudio.Services.Agent.Worker.Handlers.PowerShellExeHandler.<RunAsync>d__13.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Services.Agent.Worker.TaskRunner.<RunAsync>d__24.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Services.Agent.Worker.StepsRunner.<RunStepAsync>d__1.MoveNext()
2018-05-14T05:29:23.6133792Z ##[section]Finishing: Apply Key Vault On Config

**Comment by sphibbs** [comment]
duplicate of https://github.com/Azure/azure-powershell/issues/6165

**Comment by markcowl** [comment]
closing as duplicate
